server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -983,13 +985,12 @@ public void finalizeSnapshot(final ShardGenerations shardGenerations,
             e -> listener.onFailure(new SnapshotException(metadata.name(), snapshotId, "failed to update snapshot in repository", e));
 
         final Executor executor = threadPool.executor(ThreadPool.Names.SNAPSHOT);
-
         final boolean writeIndexGens = SnapshotsService.useIndexGenerations(repositoryMetaVersion);
-
-        final StepListener<RepositoryData> repoDataListener = new StepListener<>();
-        executor.execute(ActionRunnable.wrap(repoDataListener, this::getRepositoryData));
-        repoDataListener.whenComplete(existingRepositoryData -> {
-
+        getRepositoryData().whenComplete((existingRepositoryData, err) -> {
+            if (err != null) {
+                onUpdateFailure.accept(Exceptions.toException(err));
+                return;
+            }
```

Comment:
Given that `getRepositoryData` already dispatches to the generic thread-pool and pretty much everything within the `whenComplete` also dispatches to the executor, I removed the `executor.execute` here. Seems the extra context switch isn't worth it.